rfc/helm.md

```diff
@@ -0,0 +1,131 @@
+Helm Chart Support
+====================
+
+The current CLI install process uses Helm libraries under the hood, but just as a template library. There is currently a Helm chart under the `charts` directory that allows installing the control plane, but uninjected, and there's no `values.yaml` file provided.
+                                                                                                                                                                                                                                             
+The intention here is to provide a new chart with an injected control plane so that users can install linkerd through a simple `helm install incubator/linkerd2` command.
+                                                                                                                                                                                                                                             
+Helm charts can't rely on go code besides the functions provided by the go template library, the Sprig library, and a few extra functions provided by Helm itself. This implies a few compromises:
+                                                         
+- We can't validate the install options provided in `values.yaml`. Instead, a new set of `linkerd check` checks could help catching invalid options, post-install.
```

Comment:
I really like this idea and hadn’t thought of it myself! We could have a CLI command that consumes a values.yaml file and runs checks on it `linkerd check chart` or something like that. It could be run pre-install and maybe even highlight changes between versions (as I’m sure we’ll be mutating the values pretty extensively).

Reply:
Will this check run through the existing `install` validation code; i.e. validate the values in `values.yaml` as if they are CLI options (so a bit like `linkerd install --values values.yaml`)? I'd imagine is that plus some additional validation code, since it'd be a 1:1 mapping between the values and the CLI options.

Reply:
Yeah that sounds like it; the more code we can share here, the better :+1: